### PR TITLE
cortexm: Hard fault: Try to output as much as possible even with corrupt stack

### PR DESCRIPTION
**Pull request by gebart**
This PR reorders the DEVELHELP hardfault handler to show as much informations as possible also when the stack has been corrupted.

When the stack has been corrupted it will still be possible to read the extended fault status registers since they are hardware registers not located in RAM.

Also moved the BKPT instruction at the end to outside the condition so that the debugger halts regardless of the state of the stack.

### CODE REVIEW

File: cpu/cortexm_common/vectors_cortexm.c [hunk 1]
```diff
@@ -203,29 +217,17 @@ __attribute__((used)) void hard_fault_handler(uint32_t* sp, uint32_t corrupted,
     if( corrupted ) {
         puts("Stack pointer corrupted, reset to top of stack");
     } else {
```
Comment (OlegHahm) [inline cpu/cortexm_common/vectors_cortexm.c]: I know it's unrelated but can you adapt the three lines above to the coding conventions while you're at it?

**Comment by OlegHahm** [comment]
Untested ACK

File: cpu/cortexm_common/vectors_cortexm.c (hunk 1 above)
Comment (gebart) [inline cpu/cortexm_common/vectors_cortexm.c]: addressed, amended

**Comment by gebart** [comment]
Updated to fix a coding convention mistake in the original implementation

**Comment by gebart** [comment]
Does this go in 2016.07 or 2016.10?

**Comment by Kijewski** [comment]
It's more of a fix than a new feature, so I'd vote for 2016.07.

**Comment by kYc0o** [comment]
Me too I think it should come in 2016.07. Putting the flag now.

**Comment by kYc0o** [comment]
well, let's merge it since Murdock is ok.

**Pull request merged**